modules/apps/portal-vulcan/portal-vulcan-api/src/main/java/com/liferay/portal/vulcan/util/OpenAPIUtil.java

```diff
@@ -167,34 +169,46 @@ private static boolean _hasOKResponseContentSchemaReferenceLike(
 			return false;
 		}
 
-		Set<Map.Entry<ResponseCode, Response>> entries = responses.entrySet();
-
-		Stream<Map.Entry<ResponseCode, Response>> stream = entries.stream();
-
-		return stream.filter(
-			entry -> _isOKResponseCode(entry.getKey())
-		).map(
-			Map.Entry::getValue
-		).map(
-			Response::getContent
-		).map(
-			Map::entrySet
-		).flatMap(
-			Set::stream
-		).map(
-			Map.Entry::getValue
-		).anyMatch(
-			content -> Optional.ofNullable(
-				content.getSchema()
-			).map(
-				Schema::getReference
-			).map(
-				reference -> StringUtil.equals(
-					name, reference.substring(reference.lastIndexOf('/') + 1))
-			).orElse(
-				false
-			)
-		);
+		boolean hasOKResponse = false;
+
+		for (Map.Entry<ResponseCode, Response> entry : responses.entrySet()) {
+			if (!_isOKResponseCode(entry.getKey())) {
+				continue;
+			}
+
+			Response response = entry.getValue();
+
+			Map<String, Content> contentMap = response.getContent();
+
+			for (Map.Entry<String, Content> contentEntry :
+					contentMap.entrySet()) {
+
+				Content content = contentEntry.getValue();
+
+				if (Optional.ofNullable(
+						content.getSchema()
+					).map(
+						Schema::getReference
+					).map(
+						reference -> StringUtil.equals(
+							name,
+							reference.substring(reference.lastIndexOf('/') + 1))
+					).orElse(
+						false
+					)) {
+
+					hasOKResponse = true;
```

Comment:
@julschong I think you can return true here, and return false at line 211, so you don't need a boolean flag.

BTW this pull will only deal with `Stream<>` usages?

Reply:
Hi Dante,
I'll take a look at just returning true
yes on this pull only doing Stream. If you would like, I can add on Optional

Reply:
@dantewang you're absolutely right about returning true, I wanted to do Optional separately but I think all in one pull may be better organization in general. I will add Optional on